hazelcast/src/main/java/com/hazelcast/aggregation/impl/DistinctValuesAggregator.java

```diff
@@ -48,4 +52,34 @@ public void combine(Aggregator aggregator) {
     public Set<R> aggregate() {
         return values;
     }
+
+    @Override
+    public int getFactoryId() {
+        return AggregatorDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getId() {
+        return AggregatorDataSerializerHook.DISTINCT_VALUES;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(attributePath);
+        out.writeInt(values.size());
+        for (Object value : values) {
+            out.writeObject(value);
```

Comment:
Type of the value object can only be one of IDS, Portable, or custom but the I can not restrict this in the API. Is there a way to indicate that R should implement one of these interfaces?